examples/go/tagging/handler/handler.go

```diff
@@ -0,0 +1,244 @@
+
+		if debug() {
+			log.Println(wrapErr)
+		}
+
+		return handler.Response{
+			Body:       []byte(wrapErr.Error()),
+			StatusCode: http.StatusInternalServerError,
+		}, wrapErr
+	}
+
+	message := fmt.Sprintf("%v was tagged with %v", moRef.Value, cfg.Tag.URN)
+	log.Println(message)
+
+	return handler.Response{
+		Body:       []byte(message),
+		StatusCode: http.StatusOK,
+	}, nil
+}
+
+// vsConnect connects to vSphere govmomi API using information from vcconfig.toml.
+func vsConnect(ctx context.Context, cfg *vcConfig) error {
+	lock.Lock()
+	defer lock.Unlock()
+
+	if client == nil {
+		u := url.URL{
+			Scheme: "https",
+			Host:   cfg.VCenter.Server,
+			Path:   "sdk",
+		}
+		u.User = url.UserPassword(cfg.VCenter.User, cfg.VCenter.Password)
+		insecure := cfg.VCenter.Insecure
+
+		if debug() {
+			log.Println("connect to vSphere")
+		}
+
+		c, err := NewClient(ctx, u, insecure)
+		if err != nil {
+			return fmt.Errorf("connection to vSphere API failed: %w", err)
+		}
+
+		// Set global variable to persist connection.
+		client = c
+	}
+
+	return nil
+}
+
+func loadTomlCfg(path string) (*vcConfig, error) {
+	var cfg vcConfig
+
+	secret, err := toml.LoadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("unable to load vcconfig.toml: %w", err)
+	}
+
+	err = secret.Unmarshal(&cfg)
+	if err != nil {
+		return nil, fmt.Errorf("unable to unmarshal vcconfig.toml: %w", err)
+	}
+
+	err = validateConfig(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("insufficient information in vcconfig.toml: %w", err)
+	}
+
+	return &cfg, nil
+}
+
+// ValidateConfig ensures the bare minimum of information is in the config file.
+func validateConfig(cfg vcConfig) error {
+	reqFields := map[string]string{
+		"vcenter server":   cfg.VCenter.Server,
+		"vcenter user":     cfg.VCenter.User,
+		"vcenter password": cfg.VCenter.Password,
+		"tag URN":          cfg.Tag.URN,
+		"tag action":       cfg.Tag.Action,
+	}
+
+	// Multiple fields may be missing, but err on the first encountered.
+	for k, v := range reqFields {
+		if v == "" {
+			return errors.New("required field(s) missing, including " + k)
+		}
+	}
+
+	return nil
+}
+
+// Debug determines verbose logging
+func debug() bool {
+	verbose := os.Getenv("write_debug")
+
+	if verbose == "true" {
+		return true
+	}
+
+	return false
+}
+
+func parseEventMoRef(req []byte) (*types.ManagedObjectReference, error) {
+	var event incoming
+	var moRef types.ManagedObjectReference
+
+	err := json.Unmarshal(req, &event)
+	if err != nil {
+		return nil, fmt.Errorf("parsing of request failed: %w", err)
+	}
+
+	if event.Data.Vm == nil || event.Data.Vm.Vm.Value == "" {
+		return nil, errors.New("empty managed reference object")
+	}
+
+	// Fill information in the request into a govmomi type.
+	moRef.Type = event.Data.Vm.Vm.Type
+	moRef.Value = event.Data.Vm.Vm.Value
+
+	return &moRef, nil
+}
+
+func handleSignal(ctx context.Context) {
+	var sigCh = make(chan os.Signal)
+
+	signal.Notify(sigCh, syscall.SIGTERM, os.Interrupt)
```

Comment:
Just saw this is an unbuffered channel. It's recommended to make this a buffered channel (buffer of 2 in this case) -> https://golang.org/pkg/os/signal/#Notify

Reply:
I saw the linter suggest a buffer of 1, but I tested the code with no buffer and it worked. 

As you know, I am hesitant to add a buffer. However, I still don't fully understand how channels work. Because you have more experience and understanding, I'll go ahead and add that buffer of 2.

Reply:
It is to reliably guarantee that signals can be processed. In your case you pick up the signal. But there can be times where signal processing is delayed (or you receive multiple of these as you specified) and that's where signals won't make it through. So the guidelines (also per stdlib docs) is to use a buffer (at least "1", depending on the number of signals to expect).